app/src/main/java/com/kickstarter/ui/activities/ThreadActivity.kt

```diff
@@ -41,29 +47,33 @@ class ThreadActivity :
         ksString = environment().ksString()
         recyclerViewPaginator = RecyclerViewPaginator(binding.commentRepliesRecyclerView, { viewModel.inputs.nextPage() }, viewModel.outputs.isFetchingReplies(), false)
 
-        binding.commentRepliesRecyclerView.adapter = adapter
+        binding.commentRepliesRecyclerView.adapter = ConcatAdapter(adapter, repliesStatusAdapter, rootCommentAdapter)
         binding.commentRepliesRecyclerView.layoutManager = linearLayoutManager
 
         this.viewModel.outputs.getRootComment()
             .compose(bindToLifecycle())
             .observeOn(AndroidSchedulers.mainThread())
             .subscribe { comment ->
-                adapter.updateRootCommentCell(comment)
+                rootCommentAdapter.updateRootCommentCell(comment)
             }
 
         this.viewModel.outputs
             .onCommentReplies()
             .compose(bindToLifecycle())
             .observeOn(AndroidSchedulers.mainThread())
+            .filter { it.first.isNotEmpty() }
+            .doOnNext { linearLayoutManager.stackFromEnd = false }
             .subscribe {
-                this.adapter.takeData(it.first.reversed(), it.second)
+                this.repliesStatusAdapter.addViewMoreCell(it.second)
+                this.adapter.takeData(it.first.reversed())
```

Comment:
`this.adapter.takeData(it.first.reversed())` should not be removed?

Reply:
as we reverse   LinearLayoutManager
![image](https://user-images.githubusercontent.com/1075310/125497637-7df9589a-bc5d-4b23-a386-c7e830f5a151.png)